lib/passenger.rb

```diff
@@ -20,5 +22,30 @@ def get_drivers
     def add_trip(trip)
       @trips << trip
     end
+
+    def calc_total_spent
+      total_spent = 0
+      @trips.each do |trip|
+        if trip.end_time != nil
+          total_spent += trip.cost
+        else
+          total_spent
+        end
+      end
+      return total_spent.to_f
+    end
+
+    def calc_total_duration
+      total_duration = 0
+      @trips.each do |trip|
+        if trip.end_time != nil
+          total_duration += trip.duration
+        else
+          total_duration
```

Comment:
just a weird small note: if `trip.end_time != nil` is false and the code execution gets into this `else`, just calling `total_duration` won't actually do anything :) which is harmless, but not helpful.